b1ddi/resource_dns_forward_zone.go

```diff
@@ -0,0 +1,395 @@
+			"hosts": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The resource identifier.",
+			},
+
+			// The resource identifier.
+			"internal_forwarders": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The resource identifier.",
+			},
+
+			// Reverse zone network address in the following format: "ip-address/cidr".
+			// Defaults to empty.
+			// Read Only: true
+			"mapped_subnet": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Reverse zone network address in the following format: \"ip-address/cidr\".\nDefaults to empty.",
+			},
+
+			// Read-only. Zone mapping type.
+			// Allowed values:
+			//  * _forward_,
+			//  * _ipv4_reverse_.
+			//  * _ipv6_reverse_.
+			//
+			// Defaults to _forward_.
+			// Read Only: true
+			"mapping": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Read-only. Zone mapping type.\nAllowed values:\n * _forward_,\n * _ipv4_reverse_.\n * _ipv6_reverse_.\n\nDefaults to _forward_.",
+			},
+
+			// The resource identifier.
+			"nsgs": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The resource identifier.",
+			},
+
+			// The resource identifier.
+			"parent": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The resource identifier.",
+			},
+
+			// Zone FQDN in punycode.
+			// Read Only: true
+			"protocol_fqdn": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Zone FQDN in punycode.",
+			},
+
+			// Tagging specifics.
+			"tags": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Tagging specifics.",
+			},
+
+			// The timestamp when the object has been updated. Equals to _created_at_ if not updated after creation.
+			// Read Only: true
+			// Format: date-time
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The timestamp when the object has been updated. Equals to _created_at_ if not updated after creation.",
+			},
+
+			// The resource identifier.
+			"view": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The resource identifier.",
+			},
+
+			// The list of a forward zone warnings.
+			// Read Only: true
+			"warnings": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+				Description: "The list of a forward zone warnings.",
+			},
+		},
+	}
+}
+
+func resourceConfigForwardZoneCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*b1ddiclient.Client)
+
+	externalForwarders := make([]*models.ConfigForwarder, 0)
+	for _, ep := range d.Get("external_forwarders").([]interface{}) {
```

Comment:
I guess `ep` comes from a copy/paste from auth zone and external primaries. Using `ef` would be better.

Reply:
Updated